src/coincident/io/download.py

```diff
@@ -1118,3 +1118,78 @@ def fetch_ncalm_lpc_tiles(
         gdf.to_file(geojson_path, driver="GeoJSON")
 
     return gdf
+
+
+def fetch_lpc_tiles(
+    aoi: gpd.GeoDataFrame,
+    dataset_id: str,
+    datetime_str: str | None = None,
+    provider: str | None = None,
+    output_dir: str | None = None,
+) -> gpd.GeoDataFrame:
+    """
+    Unified fetch for LPC tiles across providers: USGS, NOAA, NCALM, NEON.
+
+    Parameters
+    ----------
+    aoi : gpd.GeoDataFrame
+        Area of interest.
+    dataset_id : str
+        Identifier string. Its format determines the provider if not explicitly set:
+          - NOAA: all digits, length 1-5.
+          - NEON: 4-character uppercase string (site ID).
+          - USGS: hard to determine with regex, function tries both USGS and NCALM
+          - NCALM: hard to determine with regex, function tries both USGS and NCALM
+    datetime_str : str, optional
+        For NEON provider only, a date in 'YYYY-MM-DD' format.
+    provider : str, optional
+        Override inferred provider. One of 'usgs', 'noaa', 'ncalm', 'neon'.
+    output_dir : str, optional
+        Directory to write provider-specific GeoJSON output.
+
+    Returns
+    -------
+    gpd.GeoDataFrame
+        Tile geometries with 'name', 'url', and 'geometry'.
+
+    Notes
+    -----
+    If provider is 'neon', you must supply datetime_str.
+    """
+    # Infer provider if not provided
```

Comment:
While I like the intention of making it easier on people to have less to type, I'd prefer to just make provider a required input. Much simpler to maintain too in case these ids change over time.